### PR TITLE
tools: Introducing ppchcalls.py for hcall count/latency stats.

### PR DESCRIPTION
**Pull request by planetharsh**
Migrating perf based powerpc-hcalls.py to a more enhanced ebpf variant inspired by existing bcc/tools/syscount.py.
Additional hcalls updated from Linux kernel source.

Signed-off-by: Harsh Prateek Bora (harshpb@linux.ibm.com)

### CODE REVIEW

**Comment by chenhengqi** [comment]
Please follow https://github.com/iovisor/bcc/blob/master/CONTRIBUTING-SCRIPTS.md and add a example.txt file.

**Comment by planetharsh** [comment]
> Please follow https://github.com/iovisor/bcc/blob/master/CONTRIBUTING-SCRIPTS.md and add a example.txt file.

Sure, will add, thanks!

**Comment by planetharsh** [comment]
Pull request updated with amended commit, updated as per guidelines in CONTRIBUTING-SCRIPTS.md file.

File: tests/python/test_tools_smoke.py [hunk 1]
```diff
@@ -279,6 +279,10 @@ def test_opensnoop(self):
     def test_pidpersec(self):
         self.run_with_int("pidpersec.py")
 
+    @skipUnless(kernel_version_ge(4,17), "requires kernel >= 4.17")
```
Comment (chenhengqi) [inline tests/python/test_tools_smoke.py:282]: Please annotate this test with `@mayFail`, currently, we don't have CI for ppc.

File: tools/ppchcalls_example.txt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+ppchcalls summarizes hcall counts across the system or a specific process,
+with optional latency information. It is very useful for general workload
+characterization, for example:
+
+# ./ppchcalls.py
+Tracing ppc hcalls, printing top 10... Ctrl+C to quit.
+^C[04:59:47]
+PPC HCALL                                        COUNT
+H_IPI                                               26
```
Comment (chenhengqi) [inline tools/ppchcalls_example.txt:12]: It seems that this is very similar to kvmexit tool, do them serve to the same purpose ?

Comment (chenhengqi) [reply tools/ppchcalls_example.txt:12]: Do you have some references for ppc hcalls ?

File: tests/python/test_tools_smoke.py (hunk 1 above)
Comment (planetharsh) [reply tests/python/test_tools_smoke.py:282]: Okay, will update.

File: tools/ppchcalls_example.txt (hunk 2 above)
Comment (planetharsh) [reply tools/ppchcalls_example.txt:12]: > 

It's different. More details can be found here: https://docs.kernel.org/powerpc/papr_hcalls.html

File: tests/python/test_tools_smoke.py (hunk 1 above)
Comment (planetharsh) [reply tests/python/test_tools_smoke.py:282]: patch updated, thanks.

**Comment by planetharsh** [comment]
@chenhengqi , any further review comments on updated pull request?
@brendangregg , could you please help with review/merge of this pull request?

Thanks
Harsh

**Comment by planetharsh** [comment]
Thanks @chenhengqi !

**Pull request merged**